### ISSUE TITLE
Should there be a lock_guard before this change of Task _state?

### ISSUE THREAD

**Issue by lawabider** [renestein/Rstein.AsyncCpp#42]
https://github.com/renestein/Rstein.AsyncCpp/blob/613eace161115e67bfb9322d537557c78e3495f2/RStein.AsyncCpp/Detail/Tasks/TaskHelpers.h#L168
I have been trying to trace the cause of some intermittent non-completion of tasks.

**Comment by renestein** [comment]
Yes, looks like the bug. The lock guard is missing. I will fix this. Thanks for the report and sorry for the inconvenience.
